QuizApp/App/Screens/Quiz/CategoryCell.swift

```diff
@@ -0,0 +1,52 @@
+import UIKit
+
+final class CategoryCell: UICollectionViewCell {
+
+    static let reuseIdentifier = String(describing: CategoryCell.self)
+    private var titleLabel: UILabel!
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+}
+
+extension CategoryCell {
+
+    func set(for category: Category) {
+        titleLabel.text = category.type.rawValue
+        titleLabel.textColor = category.color
+    }
+
+}
+
+// MARK: - ConstructViewsProtocol methods
+extension CategoryCell: ConstructViewsProtocol {
+
+    func createViews() {
+        titleLabel = UILabel()
+        contentView.addSubview(titleLabel)
+    }
+
+    func styleViews() {
+        contentView.backgroundColor = .clear
+
+        titleLabel.font = Fonts.sourceSansProBold16.font
+    }
+
+    func defineLayoutForViews() {
+        titleLabel.snp.makeConstraints {
+            $0.center.equalTo(contentView.safeAreaLayoutGuide)
+            $0.leading.trailing.equalTo(contentView.safeAreaLayoutGuide)
```

Comment:
I've noticed you are almost always constraining to `safeAreaLayoutGuide`, even for views in the inner hierarchy. 
In general, you should constrain to `safeAreaLayoutGuide` when layouting views directly inside root view of a view controller. It is not needed for views deeper in the view hierarchy. In fact, it can cause some "unexpected" issues in the layout. If you capture a view hierarchy on your QuizViewController and take a look at your categoryCollectionView layout, you will probably see what I'm talking about.

Reply:
Noted. Will correct it.